File: piper/extensions/chrome/src/engine/statements/logic_operation.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        let pointer = statement.pointer
        let op = pointer.op
        let value1 = await this.executeStatement(stringify(pointer.value1)) || null
        let value2 = await this.executeStatement(stringify(pointer.value2)) || null

        if (op === "&&") {
            return value1 && value2
        } else if (op === "||") {
            return value1 || value2
        }
    } catch (e) {
        onError(e)
    }
    return false
}

File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/logic_operation.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeLogicOperation(statement: JSONObject, onError: (e: Exception) -> Unit): Boolean {
    try {
        Log.d("hello", "execute_logic_operation")
        val pointer = statement.getJSONObject("pointer")
        val op = pointer.getString("op")
        val value1 = executeStatement(pointer["value1"].toString())
        val value2 = executeStatement(pointer["value2"].toString())

        if (op == "&&") {
            return value1 as Boolean && value2 as Boolean
        } else if (op == "||") {
            return value1 as Boolean || value2 as Boolean
        }
        return false
    } catch (e: Exception) {
        onError(e)
        return false
    }
}